core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -129,10 +136,9 @@ private[spark] class MemoryStore(
    *         iterator or call `close()` on it in order to free the storage memory consumed by the
    *         partially-unrolled block.
    */
-  private[storage] def putIterator(
+  private[storage] def putIteratorAsValues(
```

Comment:
In case it isn't obvious from the diff, the main change in this file is to split `putIterator` into two separate methods, `putIteratorAsValues` and `putIteratorAsBytes`.

It's possible that there's some opportunity to reduce code duplication here, but unless we can come up with an obvious and simple approach I would prefer to defer cleanup to followup patches.